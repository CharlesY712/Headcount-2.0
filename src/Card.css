.greaterThan {
  color: green;
}

.lessThan {
  color: red;
}

.card {
  width: 300px;
  min-width: 300px;
  border: 1px solid black;
  border-radius: 30px 30px 30px 30px;
  margin-bottom: 15px;
  padding: 10px;
}

.card-list {
  list-style-type: none;
  margin: auto;
  width: 45%;
}

.card-title {
  text-align: center;
  margin-top: 10px;
}

.selected {
  border: 3px solid blue;
}