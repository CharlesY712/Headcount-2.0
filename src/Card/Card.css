.greaterThan:before {
  content: "\02B06";
  color: green;
}

.lessThan:before {
  content: "\02B07";
  color: red;
}

.card {
  width: 260px;
  min-width: 220px;
  border: 1px solid #757575;
  border-radius: 30px 30px 30px 30px;
  margin-bottom: 15px;
  padding: 10px;
  background-image: url('../Images/background1.jpg');
  background-size: cover;
}

.card:hover{
  background-color: #F5F5F5;
  border: 2px solid black;
}

.card-list {
  list-style-type: none;
  margin: auto;
  width: 50%;
}

.card-title {
  text-align: center;
  margin-top: 10px;
  color: white;
}

.selected {
  border: 5px solid blue;
}