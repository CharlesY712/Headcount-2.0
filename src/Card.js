import React from 'react';
import './Card.css';

const Card = ({district, displayComparedCards, compareArray, selected}) => {
  let isSelected;
  if (compareArray) {
    isSelected = compareArray.find(stats => {
      return stats.location === district.location;
    });
  }
  const highlight = isSelected ? 'selected': '';
  const cardData = () => {
    return Object.keys(district.stats).map((year, index) => {
      const indicator = district.stats[year] > 0.5 ? 'greaterThan' : 'lessThan';
      return (
        <li className={indicator} key={index}>{ year }
          <span>: {Math.round(district.stats[year] * 100)/100}</span>
        </li>
      );
    });
  };

  return (
    <div
      onClick={displayComparedCards} 
      className={`card ${highlight} ${selected}`}>
      <h2 className='card-title'>{district.location}</h2>
      <ul className='card-list'>
        { cardData() }
      </ul>
    </div>
  );
};

export default Card;
