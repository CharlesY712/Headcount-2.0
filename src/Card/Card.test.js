import React from 'react';
import Card from './Card';
import { shallow, mount } from 'enzyme';

describe('Card', () => {
  const mockObject = {
    location: 'MEEKER RE1',
    stats:
      {
        '2004': 0,
        '2005': 0.505
      }
  };

  const mockFunction = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card displayComparedCards={mockFunction} district={mockObject}/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a class of greaterThan if the percentage is above .5', () => {
    expect(wrapper.find('.greaterThan').length).toEqual(1);
  });

  it('should invoke the display compared cards function on click', () => {
    const districtCard = wrapper.find('.card');
    districtCard.simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });
});