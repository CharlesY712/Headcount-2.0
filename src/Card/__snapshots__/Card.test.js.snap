// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    displayComparedCards={[Function]}
    district={
        Object {
            "location": "MEEKER RE1",
            "stats": Object {
              "2004": 0,
              "2005": 0.505,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="card-title"
>
          MEEKER RE1
</h2>,
        <ul
          className="card-list"
>
          <li
                    className="lessThan"
          >
                    2004
                    <span>
                              : 
                              0
                    </span>
          </li>
          <li
                    className="greaterThan"
          >
                    2005
                    <span>
                              : 
                              0.51
                    </span>
          </li>
</ul>,
      ],
      "className": "card  undefined",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "MEEKER RE1",
          "className": "card-title",
        },
        "ref": null,
        "rendered": "MEEKER RE1",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="lessThan"
>
              2004
              <span>
                            : 
                            0
              </span>
</li>,
            <li
              className="greaterThan"
>
              2005
              <span>
                            : 
                            0.51
              </span>
</li>,
          ],
          "className": "card-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                <span>
                  : 
                  0
</span>,
              ],
              "className": "lessThan",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    ": ",
                    0,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  ": ",
                  0,
                ],
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2005",
                <span>
                  : 
                  0.51
</span>,
              ],
              "className": "greaterThan",
            },
            "ref": null,
            "rendered": Array [
              "2005",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    ": ",
                    0.51,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  ": ",
                  0.51,
                ],
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="card-title"
>
            MEEKER RE1
</h2>,
          <ul
            className="card-list"
>
            <li
                        className="lessThan"
            >
                        2004
                        <span>
                                    : 
                                    0
                        </span>
            </li>
            <li
                        className="greaterThan"
            >
                        2005
                        <span>
                                    : 
                                    0.51
                        </span>
            </li>
</ul>,
        ],
        "className": "card  undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "MEEKER RE1",
            "className": "card-title",
          },
          "ref": null,
          "rendered": "MEEKER RE1",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="lessThan"
>
                2004
                <span>
                                : 
                                0
                </span>
</li>,
              <li
                className="greaterThan"
>
                2005
                <span>
                                : 
                                0.51
                </span>
</li>,
            ],
            "className": "card-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2004",
                  <span>
                    : 
                    0
</span>,
                ],
                "className": "lessThan",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      ": ",
                      0,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    ": ",
                    0,
                  ],
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2005",
                  <span>
                    : 
                    0.51
</span>,
                ],
                "className": "greaterThan",
              },
              "ref": null,
              "rendered": Array [
                "2005",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      ": ",
                      0.51,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    ": ",
                    0.51,
                  ],
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
